src/cpp/desktop/DesktopUtilsMac.mm

```diff
@@ -136,6 +138,54 @@ void toggleFullscreenMode(QMainWindow* pMainWindow)
       [pWindow toggleFullScreen:nil];
 }
 
+namespace {
+
+NSString* readSystemLocale()
+{
+   using namespace core;
+   using namespace core::algorithm;
+   using namespace core::system;
+   Error error;
+
+   // First, read all available locales so we can validate whether we've received
+   // a valid locale.
+   ProcessResult localeResult;
+   error = runCommand("/usr/bin/locale -a", ProcessOptions(), &localeResult);
+   if (error)
+      LOG_ERROR(error);
+
+   std::string allLocales = localeResult.stdOut;
+
+   // Now, try looking for the active locale using NSLocale.
+   std::string localeIdentifier = [[[NSLocale currentLocale] localeIdentifier] UTF8String];
```

Comment:
Do we need to eliminate trailing `@` components as we did previously?

Reply:
Yes, we should -- thanks for catching that.